pyiceberg/table/__init__.py

```diff
@@ -268,12 +268,10 @@ def _apply(self, updates: Tuple[TableUpdate, ...], requirements: Tuple[TableRequ
 
         return self
 
-    def _scan(self, row_filter: Union[str, BooleanExpression] = ALWAYS_TRUE) -> DataScan:
+    def _scan(self, row_filter: Union[str, BooleanExpression] = ALWAYS_TRUE, case_sensitive: bool = True) -> DataScan:
```

Comment:
I would leave this argument out since the `TableScan` is more like a builder pattern: `.scan().with_case_sensitive()`. Also, it doesn't allow to pass in a filter, columns, etc.

Reply:
Thanks Fokko :-) I've updated the PR to adjust the `case_sensitive` argument position to the end and closed the other comments. But I might need some more guidance regarding this one.

My understanding is this `Transaction._scan` method does not relate to the `TableScan`. It seems a helper method only used by the `Transaction.delete` method, which now takes the new `case_sensitive` and use it to plan files. I can modify the `Transaction.delete` method to use `DataScan` directly in order to pass `case_sensitive`, but that would then make this `Transaction._scan` redundant. Should I do it this way?

Reply:
Hey @jiakai-li Sorry for my ambiguous message, let me clarify. The `._scan()` returns a `DataScan` that accepts very few arguments because it follows a builder pattern. With also adding the arguments, we invalidate that pattern. Instead, below it should use `._scan().with_case_sensitive(case_sensitive)`. However, it looks like this ship has sailed already, and let's not block it since this is just a private method.

Reply:
Thanks for the explaination @Fokko , now I get it :-)